### PR TITLE
Fixed comments on params with attributes

### LINKED ISSUE #2585: Comments get removed for method parameter with attribute
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAXAngB0WAspgCo54C8AOgHYQQC2idARogE5gD6AdIfougAsE1WqLAAKEWOkBtADyEAgunSsAlkwCu6RAD4AumDVVs2xWgDOq4wHMIAentg6mIye1TpASk9jKNaTBsdSp0ADM3U3RFal8GZjZOHkw%2BQQQAJl9RSQDA2mMo8zArELswR2dXAo9csR9a0X884ONwyLNqSAAacCgLfnRbC1QwGV6IUGlIABIsXBHIVu7x2hmANwBDABtNZDQxvIAGHrzIAEkqJFCAZTUAL2QTwIAWFYh9FYBfJ4m3qFnSAslj9VtNNjs9qM-hAAIwg0SQfAbAAeABljIhUYgqDZBMs8jD0oc-h9RN8VpMxDM5pDINj4AB5MLoqiIa5YLaPP7rba7EYHQLpeGrACilyZLK5py2YUgJK%2BIMpCIB8zQkCYsFgW3xVLBvMhAukAGZhRBzlR%2Bqx0AAxYzbAByiAA7lsMTrpKpdvKyYruSraRqte7QeC%2BftoWBnqaoNdsBtoIgAEKIMKwViIAAKG1YGwYOlYwdEnsQ3to5NEStBNIWge1wp5EP5EYArNHILH40mU2nMbAnWxoBt%2Bhc1rBB4NYFRC7Ri6WIOXaJWzf6a5q637QwaIwA2NsdhPJ1PpgCq2FwrEHw6oo-Hakn04gYW2-TnYAXvymK7VtenDbDULyAB2Pc4wPbt0wAYS2IcLAgycSk0aB0DTB8wCfLYX2kUky19T9q2-Ndfz1RtwzyAAOEDO0PHs%2BASAto3QzCxGw%2BdcN1fCoB-etiP-Q0xAATkosCj0QOCtnvBjnxLLCFQpP0OPVQjuM3JsCWOCN21AxBFDCfM4LoOgNlQ2cZJ9OS8MBAig2U-VVMCGE4Q0-cuxE65GDUOBxKnSSMOk5jZIreTLM4pSN1s0j7KFJytJ0-M3LoDytQkiMTP8szAos1UQussKSIA%2ByTWiztFFYWBNEuAARRBXQS-NjNYL1TJw8z2OCxZQiIlSIukGEow0pFkTOMIiAEbFriEK0AHU1HgPFo2JJrWJa5UFOBXLeIjGFW36lEhpG7ERV88a03QabZoEVDtwWtLmoy1qsva9BOvC-Ket3HbBqoMI1GRBlzw2ZDWBFZFgkQCwLDvbyIzI67RBYt82JWtq1s-LrXrEGFgI%2BgAlRA4FYeAzrmiNnlh2h4ffMAl3%2BVaOpsvK%2BNEGEKJxvG03ge1NDopkzh0OhhmjGFX0p6nqTauhNC2QZXVZa000MlQ2BIeZ6Y2glBI03H8fgfBJeljE5dYBW6rbaABGzeN8w4J0ZrxYXEarZG6fWrc8iJNsBpKnNMAZVh0SsImLujGH7eWx2HpR3U0cZ2h0kc05PdYb3ff99BOe5sJecYAXNtDu6kYeiWpbUGXEEN42lcBVXXcFKLTi9jYfb9tQrF14vS-LgGTY0s2LaQthrdti688XIKI%2Bd1GXpjiB0kKhOUQANX1RNjHgWxA9QkPFoRsPl1pp7q7s6R0j6%2BfkWtcqkMhlfLnXofUNJkeP3u2lI%2BVaOI3Sbaz4q2B0AAcX4MDUG4NIYb2DmTd4AVR6ZQDKFSeDNP7vQTnrEuGJEziWgAAa0TDmbB-ALAMioNcXMoktRcyhnkRifk4bQOfgXOBOUEFq0FFjU4joXToP4P2bEytEBVW%2BlQNQE4qCKEuLRFgrAc55FSrQ9KMCX6riYVHKen8Wb11dDYKgF8qBX0nLcLRANNDpiILAC4Vx0AA0hqhahT8qZj0YeuZhNdj4aw0fmKgVi1i9hxBItgAjjDCMhtIwItjt4iwcUopxKjEF5CNOpFBxdJSolzEweAGwoKwAhjiDhpcbFSTsaLL82Vonv1UXE%2BOgRIDALYGoBgoRtgAGlECIGwOY7E6ALi4I2LooOEYwk3SWvncOjjnqxMCEaOuVSME9KwZKCwJUyqXEdFYRAOtUGl2AemUB8F6qNUGTvYZe82pcRdkfMQRo57TOzNRdMFUW7QHUAlTxOh4DHiEZOKq0BoI5hEfk3yhTIlWVKSGcpEzT5VJqU8jp2x2SlU0AhbA7JMCcn%2BUxORt0FEMKBBPGJLDjTfyqc01pA106SKZDMqgcyMQhJ6ocSBYAKYO2OQ9U5zjzmiCNMgqpcKPLoHwAgKUoSCnhN6PoJ4iwLDWghiMWxnwgA)

#### Code

```fsharp
type MyType =
    member _.MyMethod
        (
            [<MyAttribute>] inputA: string  // my input
        )
        =
        printf inputA

    member _.MyMethod2
        (
            inputA: string  // my input
        )
        =
        printf inputA

```
            

#### Result

```fsharp
type MyType =
    member _.MyMethod([<MyAttribute>] inputA: string) = printf inputA

    member _.MyMethod2
        (inputA: string) // my input
        =
        printf inputA

```
            
#### Problem description

If you have a type method with an attribute and a comment afterwards the comment gets removed. If you don't include an attribute the formatting does not remove the comment.

#### Extra information

- [ ] The formatted result breaks my code.
- [ ] The formatted result gives compiler warnings.
- [ ] I or my company would be willing to help fix this.

#### Options

Fantomas main branch at 2022-10-14T16:59:37Z - 35f093ce1b8a19d9e9ade3da2c0f855923d8a106

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://fsprojects.github.io/fantomas/docs/end-users/IgnoreFiles.html)?</sub>

### PR DESCRIPTION
**Pull request by stefan-schweiger**
Fixes #2585

Added addtional cases for  `SynPat_Attrib`, `SynPat_Named`, `SynPat_Typed` to `visitLastChildNode`

### CODE REVIEW

**Comment by nojaf** [comment]
Hi Stefan, this looks good! Would you mind adding a [CHANGELOG entry](https://fsprojects.github.io/fantomas/docs/contributors/Pull%20request%20ground%20rules.html#Changelog)?

**Comment by stefan-schweiger** [comment]
@nojaf Updated the changelog, let me know if there is anything else ;)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## [Unreleased]
+
+### Fixed
+* Comments get removed for method parameter with attribute [#2585](https://github.com/fsprojects/fantomas/issues/2585)
```
Comment (nojaf) [inline CHANGELOG.md:6]: Super nitpick but all the other entries have `.` at end of the sentence before the `[#`.
No need to update it, just something I should add to the documentation.

**Review by nojaf — APPROVED** [review]
Looks good, thanks!

**Comment by stefan-schweiger** [comment]
Perfect thanks! When do you think this will be released? This is currently the major bug which prevents us from switching to Fantomas 5

**Comment by stefan-schweiger** [comment]
Against 5.0.6, but I guess in the worst case we can live with using the beta

**Comment by nojaf** [comment]
The beta is available: https://www.nuget.org/packages/fantomas/5.1.0-beta-001
Our policy is that we no longer change the style once we have labelled it beta.
In short, this version will be very close to what `5.1` stable will be.

**Pull request merged**